autowiring/AutowirableSlot.h

```diff
@@ -68,8 +68,13 @@ class DeferrableAutowiring
   /// </remarks>
   std::weak_ptr<CoreContext> m_context;
 
-  // The next entry to be autowired in this sequence
-  DeferrableAutowiring* m_pFlink = nullptr;
+  /// The unique ptr to the registration of the singal handler which is used to finish
+  /// the autowiring. The registration can be used to cancel the signal handler.
+  std::unique_ptr<autowiring::registration_t> m_deferred_autowire;
```

Comment:
Why did you have to make this a pointer type?  Is there a limitation with `autowiring::registration_t` that prevents you from making it a member?